src/components/SystemPreferences.vue

```diff
@@ -0,0 +1,95 @@
+<script>
+  import VueSlider from 'vue-slider-component';
+  import 'vue-slider-component/theme/default.css';
+  export default {
+    components: {
+      VueSlider
+    },
+    props: {
+      memoryInGB: {
+        type: Number,
+        default: 2
+      },
+      numberCPUs: {
+        type: Number,
+        default: 2
+      },
+      availMemoryInGB: {
+          type: Number,
+          default: 0
+      },
+      availNumCPUs: {
+          type: Number,
+          default: 0
+      }
+    },
+    methods: {
+      updatedMemory(value) {
+        this.$emit('updateMemory', value);
+      },
+      updatedCPU(value) {
+        this.$emit('updateCPU', value);
+      },
+      makeArray(min, max) {
+        let size = max - min + 1;
+        return [...Array(size).keys()].map(i => i + min);
+      }
+    },
+    computed: {
+      memoryMarks: function() {
+        return this.makeArray(2, this.availMemoryInGB);
+      },
+      CPUMarks: function() {
+        return this.makeArray(1, this.availNumCPUs);
+      },
+      disableMemory: function() {
+        return this.availMemoryInGB <= 2
+      },
+      disableCPUs: function() {
+        return this.availNumCPUs <= 1
+      }
+    }
+  }
+</script>
+
+<template>
+  <div class="system-preferences">
+    <p>System Preferences:</p>
+    <div id="memoryInGBWrapper" class="labeled-slider">
+      <div class="slider-label">Memory (GB):</div>
+      <vue-slider :value="memoryInGB"
+                  :min="2"
+                  :max="availMemoryInGB"
+                  :interval="1"
+                  :marks="memoryMarks"
```

Comment:
Having marks on every possible value _and_ tooltip seems a bit excessive?

Reply:
Good call. I think the default with a tooltip and no marks isn't very useful most of the time -- better to show the user what they can do before they start an action.